2011-02-28  Aurelien Jarno  <aurelien@aurel32.net>

	* sysdeps/alpha/stackinfo.h: Define DEFAULT_STACK_PERMS with PF_X.

diff --git glibc-2.13/glibc-ports-2.13/sysdeps/alpha/stackinfo.h glibc-2.13/glibc-ports-2.13/sysdeps/alpha/stackinfo.h
index 0a281bd..d9dbc35 100644
--- glibc-2.13/glibc-ports-2.13/sysdeps/alpha/stackinfo.h
+++ glibc-2.13/glibc-ports-2.13/sysdeps/alpha/stackinfo.h
@@ -22,7 +22,13 @@
 #ifndef _STACKINFO_H
 #define _STACKINFO_H	1
 
+#include <elf.h>
+
 /* On Alpha the stack grows down.  */
 #define _STACK_GROWS_DOWN	1
 
+/* Default to an executable stack.  PF_X can be overridden if PT_GNU_STACK is
+ * present, but it is presumed absent.  */
+#define DEFAULT_STACK_PERMS (PF_R|PF_W|PF_X)
+
 #endif	/* stackinfo.h */

